rclpy/src/rclpy/_rclpy.c

```diff
@@ -3369,6 +3370,385 @@ rclpy_clock_set_ros_time_override(PyObject * Py_UNUSED(self), PyObject * args)
+          PyErr_Format(PyExc_RuntimeError, "Failed to parse yaml params file: %s", param_files[i]);
+          successful = false;
+        } else {
+          if (!_populate_node_parameters_from_rcl_params(
+              params, allocator, parameter_cls, parameter_type_cls, params_by_node_name))
+          {
+            PyErr_Format(PyExc_RuntimeError,
+              "Failed to fill params dict from file: %s", param_files[i]);
+            rcl_yaml_node_struct_fini(params);
+            return false;
+          }
+        }
+        rcl_yaml_node_struct_fini(params);
+      }
+      allocator.deallocate(param_files[i], allocator.state);
+    }
+    allocator.deallocate(param_files, allocator.state);
+  }
+  return successful;
+}
+
+/// Get a list of parameters for the current node from rcl_yaml_param_parser
+/**
+ * On failure, an exception is raised and NULL is returned if:
+ *
+ * Raises ValueError if the argument is not a node handle.
+ * Raises RuntimeError if the parameters file fails to parse
+ *
+ * \param[in] parameter_cls The rclpy.parameter.Parameter class object.
+ * \param[in] node_handle Capsule pointing to the node handle
+ * \return NULL on failure
+ *         A list of rclpy.parameter.Parameter on success (may be empty).
+ *
+ */
+static PyObject *
+rclpy_get_node_parameters(PyObject * Py_UNUSED(self), PyObject * args)
+{
+  PyObject * parameter_cls;
+  PyObject * node_capsule;
+  if (!PyArg_ParseTuple(args, "OO", &parameter_cls, &node_capsule)) {
+    return NULL;
+  }
+
+  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(node_capsule, "rcl_node_t");
+  if (!node) {
+    return NULL;
+  }
+
+  PyObject * params_by_node_name = PyDict_New();
+  if (NULL == params_by_node_name) {
+    return NULL;
+  }
+
+  if (!PyObject_HasAttrString(parameter_cls, "Type")) {
+    PyErr_Format(PyExc_RuntimeError, "Parameter class is missing 'Type' attribute");
+    Py_DECREF(params_by_node_name);
+    return NULL;
+  }
+  PyObject * parameter_type_cls = PyObject_GetAttrString(parameter_cls, "Type");
+  if (NULL == parameter_type_cls) {
+    /* PyObject_GetAttrString raises AttributeError on failure. */
+    Py_DECREF(params_by_node_name);
+    return NULL;
+  }
+
+  const rcl_node_options_t * node_options = rcl_node_get_options(node);
+  const rcl_allocator_t allocator = node_options->allocator;
+
+  if (node_options->use_global_arguments) {
+    if (!_parse_param_files(rcl_get_global_arguments(), allocator, parameter_cls,
+      parameter_type_cls, params_by_node_name))
+    {
+      Py_DECREF(parameter_type_cls);
+      Py_DECREF(params_by_node_name);
+      return NULL;
+    }
+  }
+
+  if (!_parse_param_files(&(node_options->arguments), allocator, parameter_cls,
+    parameter_type_cls, params_by_node_name))
+  {
+    Py_DECREF(parameter_type_cls);
+    Py_DECREF(params_by_node_name);
+    return NULL;
+  }
+  Py_DECREF(parameter_type_cls);
+
+  const char * node_namespace = rcl_node_get_namespace(node);
```

Comment:
is this deallocated anywhere?

Reply:
No. Since this is a `const` "borrowed" reference to the internal node string. It's lifetime is bound to the node it is from.